### PR TITLE
feat!: added typesafe environment variables

### LINKED ISSUE #242: Creating typesafe environment variables for better developer experience
## Description
As of right now, the environment variable in the leaderboard is not typesafe that could cause problems related to developer experience. Such as invalid variables are not detected and required variable are not known until runtime.

## Solution
Using a schema validator validate the current environment variables and apply changes throughout the codebase.

## Willing to solve the issue or working on it?
Yes

### PR DESCRIPTION
**Pull request by Re-Dei**
## Issue Fixed:
Fixed #242 

## Changes:
- added @t3-oss/env-nextjs for env validation as well as the required Zod package
- env.mjs defines all the required environment variables, which are classified into server and client environment variables, in an object env that is exported to the files that require it
- changed all process.env.[variable_name] -> env.variable names where env is the import env object from env.mjs

## Problem solved:
The project used to get build when a required environment variable was missing which could only be found on runtime. Now the project fails to build if a required environment is missing. An example is given below:
Screenshot 2024-02-25 170513

## Testing:
Runs without an issue on development but fails in production due to missing env GITHUB_PAT. Checked on WSL2.

### CODE REVIEW

File: env.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { createEnv } from "@t3-oss/env-nextjs";
+import { z } from "zod";
+
+export const env = createEnv({
+  server: {
+    GITHUB_PAT: z.string(),
+    GITHUB_TOKEN: z.string().optional(),
```
[the remark this answers is not in the document]
Comment (Re-Dei) [reply env.mjs:7]: Yes

[the remark this answers is not in the document]
Comment (rithviknishad) [reply env.mjs:7]: Where? I don't see 👀

[the remark this answers is not in the document]
Comment (Re-Dei) [reply env.mjs:7]: I believe it was in scripts/generateNewContributors.js

[the remark this answers is not in the document]
Comment (rithviknishad) [reply env.mjs:7]: If you meant about the python code, that's part of the scrapper workflow and not part of the NextJS project.

[the remark this answers is not in the document]
Comment (Re-Dei) [reply env.mjs:7]: So should I remove it?

[the remark this answers is not in the document]
Comment (Re-Dei) [reply env.mjs:7]: I will remove it asap.

File: env.mjs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    NEXT_PUBLIC_META_IMG: process.env.NEXT_PUBLIC_META_IMG,
+    NEXT_PUBLIC_META_DESCRIPTION: process.env.NEXT_PUBLIC_META_DESCRIPTION,
+    NEXT_PUBLIC_META_URL: process.env.NEXT_PUBLIC_META_URL,
+    NEXT_PUBLIC_PAGE_TITLE: process.env.NEXT_PUBLIC_PAGE_TITLE,
+    NEXT_PUBLIC_CONTRIBUTORS_INFO: process.env.NEXT_PUBLIC_CONTRIBUTORS_INFO,
+    GITHUB_PAT:
+      process.env.NODE_ENV === "development" && !process.env.GITHUB_PAT
+        ? ""
+        : process.env.GITHUB_PAT,
+    GITHUB_TOKEN: process.env.GITHUB_TOKEN,
```
[the remark this answers is not in the document]
Comment (Re-Dei) [reply env.mjs]: I have removed the GITHUB_TOKEN in the latest commit

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**